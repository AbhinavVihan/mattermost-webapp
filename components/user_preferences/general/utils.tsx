// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import { defineMessages } from "react-intl";
import { t } from "utils/i18n";

type setting = {
    hasError: boolean;
    errors: {
        emailError?: string;
        clientError: string;
        serverError: string;
    };
    value: string | undefined;
    isDisabled?: boolean;
};

export type Settings = {
    [key: string]:
        | {
              hasError?: boolean;
              errors?: {
                  emailError?: string;
                  clientError?: string;
                  serverError?: string;
              };
              value?: string;
              isDisabled?: boolean;
          }
        | undefined;
    username: setting;
    firstName: setting;
    lastName: setting;
    // fullName: setting;
    nickname: setting;
    position: setting;
    originalEmail: setting;
    email: setting;
    confirmEmail: setting;
    currentPassword: setting;
};

export const holders = defineMessages({
    usernameReserved: {
        id: t("user.settings.general.usernameReserved"),
        defaultMessage: "This username is reserved, please choose a new one.",
    },
    usernameGroupNameUniqueness: {
        id: t("user.settings.general.usernameGroupNameUniqueness"),
        defaultMessage: "This username conflicts with an existing group name.",
    },
    usernameRestrictions: {
        id: t("user.settings.general.usernameRestrictions"),
        defaultMessage:
            "Username must begin with a letter, and contain between {min} to {max} lowercase characters made up of numbers, letters, and the symbols '.', '-', and '_'.",
    },
    validEmail: {
        id: t("user.settings.general.validEmail"),
        defaultMessage: "Please enter a valid email address.",
    },
    emailMatch: {
        id: t("user.settings.general.emailMatch"),
        defaultMessage: "The new emails you entered do not match.",
    },
    incorrectPassword: {
        id: t("user.settings.general.incorrectPassword"),
        defaultMessage: "Your password is incorrect.",
    },
    emptyPassword: {
        id: t("user.settings.general.emptyPassword"),
        defaultMessage: "Please enter your current password.",
    },
    validImage: {
        id: t("user.settings.general.validImage"),
        defaultMessage:
            "Only BMP, JPG, JPEG, or PNG images may be used for profile pictures",
    },
    imageTooLarge: {
        id: t("user.settings.general.imageTooLarge"),
        defaultMessage: "Unable to upload profile image. File is too large.",
    },
    uploadImage: {
        id: t("user.settings.general.uploadImage"),
        defaultMessage: "Click 'Edit' to upload an image.",
    },
    uploadImageMobile: {
        id: t("user.settings.general.mobile.uploadImage"),
        defaultMessage: "Click to upload an image",
    },
    fullName: {
        id: t("user.settings.general.fullName"),
        defaultMessage: "Full Name",
    },
    nickname: {
        id: t("user.settings.general.nickname"),
        defaultMessage: "Nickname",
    },
    username: {
        id: t("user.settings.general.username"),
        defaultMessage: "Username",
    },
    profilePicture: {
        id: t("user.settings.general.profilePicture"),
        defaultMessage: "Profile Picture",
    },
    close: {
        id: t("user.settings.general.close"),
        defaultMessage: "Close",
    },
    position: {
        id: t("user.settings.general.position"),
        defaultMessage: "Position",
    },
});

export const yourInfo = {
    id: t("user.settings.general.yourInfo"),
    defaultMessage: "Your information",
};
