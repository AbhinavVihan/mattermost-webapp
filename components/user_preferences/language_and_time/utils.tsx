// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {t} from 'utils/i18n';
import {localizeMessage} from 'utils/utils';

import {FieldsetCheckbox} from 'components/widgets/modals/generic/checkbox-item-creator';
import {FieldsetRadio} from 'components/widgets/modals/generic/radio-item-creator';
import {
    FieldsetReactSelect,
    Limit,
} from 'components/widgets/modals/generic/react-select-item-creator';

import {CategorySorting} from '@mattermost/types/channel_categories';
import * as I18n from 'i18n/i18n.jsx';

export type SelectedOption = {
    value: string;
    label: string;
};

export const showLanguageTitle = {
    id: t('user.settings.languageAndTime.language.title'),
    defaultMessage: 'Language',
};

export const showLanguageTitleDescription = {
    id: t('user.settings.languageAndTime.language.title.description'),
    defaultMessage: "Select the language you'd prefer to use in Mattermost.",
};

export const languageSelectDescription = {
    id: t('user.settings.languageAndTime.languageSelect.description'),
    defaultMessage:
        'Join the Mattermost Translation Server to help with translations.',
};

export const languageLabels: Limit[] = [];

const locales: any = I18n.getLanguages();

const languages = Object.keys(locales).
    map((l) => {
        return {
            value: locales[l].value,
            label: locales[l].name,
            order: locales[l].order,
        };
    }).
    sort((a, b) => a.order - b.order);

languages.forEach((lang) => {
    languageLabels.push({value: lang.value, label: lang.label});
});

export const showTimezoneTitle = {
    id: t('user.settings.languageAndTime.Timezone.title'),
    defaultMessage: 'Timezone',
};

export const showTimezoneDescription = {
    id: t('user.settings.languageAndTime.Timezone.Description'),
    defaultMessage:
        'Select your timezone for message timestamps and email notifications.',
};

export const showAutoTimezoneCheckboxTitle = {
    id: t('user.settings.languageAndTime.setAutomaticTimezone'),
    defaultMessage: 'Set timezone automatically',
};

export const showTimeFormatTitle = {
    id: t('user.settings.languageAndTime.TimeFormat.title'),
    defaultMessage: 'Time format',
};

export const showTimeFormatDescription = {
    id: t('user.settings.languageAndTime.TimeFormat.description'),
    defaultMessage: 'Select the time format you prefer.',
};

const twelveHourFormat = '12-hour clock (example: 4:00 PM)';
const twentyFourHourFormat = '24-hour clock (example: 16:00)';

export const TimeFormatData: FieldsetRadio = {
    options: [
        {
            dataTestId: `${twelveHourFormat}-test-id`,
            title: {
                id: 'user.settings.languageAndTime.TimeFormat.12Hour',
                defaultMessage: '12-hour clock (example: 4:00 PM)',
            },
            name: `${twelveHourFormat}-clockFormat}`,
            key: `${twelveHourFormat}-key`,
            value: 'false',
        },
        {
            dataTestId: `${twentyFourHourFormat}-test-id`,
            title: {
                id: 'user.settings.languageAndTime.TimeFormat.24Hour',
                defaultMessage: '24-hour clock (example: 16:00)',
            },
            name: `${twentyFourHourFormat}-clockFormat`,
            key: `${twelveHourFormat}-key`,
            value: 'true',
        },
    ],
};

export const dMsTitle = {
    id: t('user.settings.sidebar.direct.title'),
    defaultMessage: 'Direct messages',
};

export const channelsTitle = {
    id: t('user.settings.sidebar.channels.title'),
    defaultMessage: 'Channel categories',
};

export const unreadsInputFieldData: FieldsetCheckbox = {
    title: {
        id: 'user.settings.sidebar.showUnreadsCategoryTitle',
        defaultMessage: 'Group unread channels separately',
    },
    name: 'showUnreadsCategory',
    dataTestId: 'showUnreadsCategoryOn',
};

export const unreradsDescription = {
    id: t('user.settings.sidebar.showUnreadsCategoryDesc'),
    defaultMessage:
        'When enabled, all unread channels and direct messages will be grouped together in the sidebar.',
};

export const dmSortTitle = {
    id: 'user.settings.sidebar.DMsSorting',
    defaultMessage: 'Sort direct messages',
};

export const channelsSortTitle = {
    id: 'user.settings.sidebar.channelsSorting.title',
    defaultMessage: 'Default sort for channel categories',
};

export const channelsSortDesc = {
    id: 'user.settings.sidebar.channelsSorting.desc',
    defaultMessage:
        "You can override sorting for individual categories using the category's sidebar menu.",
};

export const sortInputFieldData: FieldsetRadio = {
    options: [
        {
            dataTestId: `dmSorting-${CategorySorting.Recency}`,
            title: {
                id: 'user.settings.sidebar.DMsSorting.alphabetically',
                defaultMessage: 'Alphabetically',
            },
            name: `dmSorting-${CategorySorting.Recency}`,
            key: `dmSorting-${CategorySorting.Alphabetical}`,
            value: CategorySorting.Alphabetical,
        },
        {
            dataTestId: `dmSorting-${CategorySorting.Recency}`,
            title: {
                id: 'user.settings.sidebar.DMsSorting.recent',
                defaultMessage: 'By recent activity',
            },
            name: `dmSorting-${CategorySorting.Recency}`,
            key: `dmSorting-${CategorySorting.Recency}`,
            value: CategorySorting.Recency,
        },
    ],
};

export const channelsSortInputFieldData: FieldsetRadio = {
    options: [
        {
            dataTestId: `channelsSorting-${CategorySorting.Recency}`,
            title: {
                id: 'user.settings.sidebar.channelsSorting.alphabetically',
                defaultMessage: 'Alphabetically',
            },
            name: `channelsSorting-${CategorySorting.Alphabetical}`,
            key: `channelsSorting-${CategorySorting.Alphabetical}`,
            value: CategorySorting.Alphabetical,
        },
        {
            dataTestId: `channelsSorting-${CategorySorting.Recency}`,
            title: {
                id: 'user.settings.sidebar.channelsSorting.recent',
                defaultMessage: 'By recent activity',
            },
            name: `channelsSorting-${CategorySorting.Recency}`,
            key: `channelsSorting-${CategorySorting.Recency}`,
            value: CategorySorting.Recency,
        },
        {
            dataTestId: `channelsSorting-${CategorySorting.Manual}`,
            title: {
                id: 'user.settings.sidebar.channelsSorting.manual',
                defaultMessage: 'Manually',
            },
            name: `channelsSorting-${CategorySorting.Manual}`,
            key: `channelsSorting-${CategorySorting.Manual}`,
            value: CategorySorting.Manual,
        },
    ],
};

export const languageSelectInputFieldData: FieldsetReactSelect = {
    options: languageLabels,
};

export const dmLimitTitle = {
    id: 'user.settings.sidebar.limitVisibleGMsDMsTitle',
    defaultMessage: 'Number of direct messages to show in the channel sidebar',
};

export const dmLimitDescription = {
    id: t('user.settings.sidebar.limitVisibleGMsDMsDesc'),
    defaultMessage:
        'You can change direct message settings using the direct messages sidebar menu.',
};
