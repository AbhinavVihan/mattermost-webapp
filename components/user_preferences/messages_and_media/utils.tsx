import { FieldsetRadio } from "components/widgets/modals/generic/radio-item-creator";
import Constants from "utils/constants";
import { t } from "utils/i18n";

const Preferences = Constants.Preferences;

// interface messageAndMediaSettingsType {
//     [Preferences.COLLAPSED_REPLY_THREADS]: string,

// }

// type MessageAndMediaSettingsType = {
//     [key: string]: string | undefined;
//     [Preferences.COLLAPSED_REPLY_THREADS]: string;
//     [Preferences.NAME_NAME_FORMAT]: string;
//     [Preferences.AVAILABILITY_STATUS_ON_POSTS]: string;
//     [Preferences.COLLAPSE_DISPLAY]: string;
//     [Preferences.LINK_PREVIEW_DISPLAY]: string;
//     [Preferences.ONE_CLICK_REACTIONS_ENABLED]: string;
// }

export const messageDisplayTtle = {
    id: t("user.settings.messagesAndMedia.messageDisplay.title"),
    defaultMessage: "Message display",
};

export const messageDisplayDescription = {
    id: t("user.settings.messagesAndMedia.messageDisplay.description"),
    defaultMessage: "Here’s how your messages will look",
};

export const messageDensityTitle = {
    id: t("user.settings.messagesAndMedia.messageDensity.title"),
    defaultMessage: "Message Density",
};

export const MessageDensityData: FieldsetRadio = {
    options: [
        {
            dataTestId: `messageDensity-test-id-clean`,
            title: {
                id: "user.settings.messagesAndMedia.messageDensity.clean",
                defaultMessage: "Clean: Easy to scan and read",
            },
            name: `messageDensity-clean`,
            key: `messageDensity-key-clean`,
            value: Preferences.MESSAGE_DISPLAY_CLEAN,
        },
        {
            dataTestId: `messageDensity-test-id-compact`,
            title: {
                id: "user.settings.messagesAndMedia.messageDensity.compact",
                defaultMessage:
                    "Compact: Fit as many messages on-screen as possible",
            },
            name: `messageDensity-compact`,
            key: `messageDensity-key-compact`,
            value: Preferences.MESSAGE_DISPLAY_COMPACT,
        },
    ],
};

export const messageWidthTitle = {
    id: t("user.settings.messagesAndMedia.messageWidth.title"),
    defaultMessage: "Message width",
};

export const MessageWidthData: FieldsetRadio = {
    options: [
        {
            dataTestId: `messageWidth-test-id-full-width`,
            title: {
                id: "user.settings.messagesAndMedia.messageWidth.full-width",
                defaultMessage: "Full-width",
            },
            name: `messageWidth-full-width`,
            key: `messageWidth-key-full-width`,
            value: Preferences.CHANNEL_DISPLAY_MODE_FULL_SCREEN,
        },
        {
            dataTestId: `messageWidth-test-id-centered`,
            title: {
                id: "user.settings.messagesAndMedia.messageWidth.centered",
                defaultMessage: "Fixed-width, centered",
            },
            name: `messageWidth-centered`,
            key: `messageWidth-key-centered`,
            value: Preferences.CHANNEL_DISPLAY_MODE_CENTERED,
        },
    ],
};

export const teamMateNameDisplayTitle = {
    id: t("user.settings.messagesAndMedia.teammatename.title"),
    defaultMessage: "Teammate name display",
};

export const teammateNameData: FieldsetRadio = {
    options: [
        {
            dataTestId: `teammateName-test-id-username`,
            title: {
                id: "user.settings.messagesAndMedia.teammateName.username",
                defaultMessage: "Show username",
            },
            name: `teammateName-username`,
            key: `teammateName-key-username`,
            value: Constants.TEAMMATE_NAME_DISPLAY.SHOW_USERNAME,
        },
        {
            dataTestId: `teammateName-test-id-firstAndLast`,
            title: {
                id: "user.settings.messagesAndMedia.teammateName.firstAndLast",
                defaultMessage: "Show first and last name",
            },
            name: `teammateName-firstAndLast`,
            key: `teammateName-key-firstAndLast`,
            value: Constants.TEAMMATE_NAME_DISPLAY.SHOW_FULLNAME,
        },
        {
            dataTestId: `teammateName-test-id-nickname`,
            title: {
                id: "user.settings.messagesAndMedia.teammateName.nickname",
                defaultMessage:
                    "Show nickname, otherwise show first and last name",
            },
            name: `teammateName-nickname`,
            key: `teammateName-key-nickname`,
            value: Constants.TEAMMATE_NAME_DISPLAY.SHOW_NICKNAME_FULLNAME,
        },
    ],
};

export const profileStatusTitle = {
    id: t("user.settings.messagesAndMedia.profileStatus.title"),
    defaultMessage: "Online status on profile images",
};

export const profileStatusData = {
    dataTestId: "profileStatus-test-id",
    title: {
        id: "user.settings.messagesAndMedia.profileStatus",
        defaultMessage: "Show online status on profile images",
    },
    name: "profileStatus-name",
    key: "profileStatus-key",
};

export const threadsTitle = {
    id: t("user.settings.messagesAndMedia.threads.title"),
    defaultMessage: "Threads",
};
export const enableCRTData = {
    dataTestId: "enableCRT-test-id",
    title: {
        id: "user.settings.messagesAndMedia.enableCRT.title",
        defaultMessage: "Enabled collapsed replies (Beta)",
    },
    description: {
        id: "user.settings.messagesAndMedia.enableCRT.description",
        defaultMessage:
            "        When enabled, replies aren't shown in the channel. You'll be notified about threads you're following in the Threads view. Please review our documentation for known issues and help provide feedback in our community channel.",
    },
    name: "enableCRT-name",
    key: "enableCRT-key",
};

export const openThreadsData = {
    dataTestId: "openThreads-test-id",
    title: {
        id: "user.settings.messagesAndMedia.openThreads",
        defaultMessage: "Click to open threads",
    },
    description: {
        id: "user.settings.messagesAndMedia.openThreads.description",
        defaultMessage:
            "    When enabled, select any part of a message to open the reply thread.",
    },
    name: "openThreads-name",
    key: "openThreads-key",
};

export const previewsTitle = {
    id: t("user.settings.messagesAndMedia.previewsTitle.title"),
    defaultMessage: "Image and link previews",
};

export const imagePreviewsData = {
    dataTestId: "imagePreview-test-id",
    title: {
        id: "user.settings.messagesAndMedia.imagePreview.title",
        defaultMessage: "Show expanded image previews",
    },
    name: "imagePreview-name",
    key: "imagePreview-key",
};

export const websitePreviewsData = {
    dataTestId: "websitePreview-test-id",
    title: {
        id: "user.settings.messagesAndMedia.websitePreview.title",
        defaultMessage: "Show website link previews",
    },
    name: "websitePreview-name",
    key: "websitePreview-key",
};

export const quickReactionTitle = {
    id: t("user.settings.messagesAndMedia.quickReaction.title"),
    defaultMessage: "Quick reactions",
};

export const quickReactionDescription = {
    id: "user.settings.messagesAndMedia.quickReaction.description",
    defaultMessage:
        "React quickly with recent or frequent reactions when hovering over a message.",
};

export const quickReactionCheckboxData = {
    dataTestId: "quickReactionCheckbox-test-id",
    title: {
        id: "user.settings.messagesAndMedia.quickReactionCheckbox.title",
        defaultMessage: "Enable quick reactions on messages",
    },
    name: "quickReactionCheckbox-name",
    key: "quickReactionCheckbox-key",
};

export const quickReactionRadioTitle = {
    id: "user.settings.messagesAndMedia.quickReactionRadio.title",
    defaultMessage: "Which reactions would you prefer to show?",
};

export const quickReactionRadioData: FieldsetRadio = {
    options: [
        {
            dataTestId: `quickReactionRadio-test-id-frequent`,
            title: {
                id: "user.settings.messagesAndMedia.quickReactionRadio.frequent.title",
                defaultMessage: "Show my most frequently used reactions",
            },
            name: `quickReactionRadio-frequent`,
            key: `quickReactionRadio-key-frequent`,
            value: Constants.TEAMMATE_NAME_DISPLAY.SHOW_USERNAME,
        },
        {
            dataTestId: `quickReaction-test-id-preffered`,
            title: {
                id: "user.settings.messagesAndMedia.quickReaction.preffered",
                defaultMessage: "Show my preferred reactions",
            },
            name: `quickReaction-preffered`,
            key: `quickReaction-key-preffered`,
            value: Constants.TEAMMATE_NAME_DISPLAY.SHOW_FULLNAME,
        },
    ],
};

export const reactionsDemoTitle = {
    id: t("user.settings.messagesAndMedia.reactionsDemo.title"),
    defaultMessage: "Here’s how your reactions will look",
};
