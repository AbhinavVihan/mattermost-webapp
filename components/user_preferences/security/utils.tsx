// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import AccessHistoryModal from "components/access_history_modal";
import ActivityLogModal from "components/activity_log_modal";
import LocalizedIcon from "components/localized_icon";
import ToggleModalButton from "components/toggle_modal_button";
import React, { useCallback, useState } from "react";
import {
    defineMessages,
    FormattedDate,
    FormattedMessage,
    FormattedTime,
} from "react-intl";
import { t } from "utils/i18n";

type setting = {
    hasError: boolean;
    errors: {
        clientError: string;
        serverError: string;
    };
    value: string | undefined;
};

export type PasswordSettings = {
    [key: string]:
        | {
              hasError?: boolean;
              errors?: {
                  clientError?: string;
                  serverError?: string;
              };
              value?: string;
          }
        | undefined;
    confirmNewPassword: setting;
    newPassword: setting;
    currentPassword: setting;
};

export const passwordTitle = {
    id: t("user.settings.security.password"),
    defaultMessage: "Password",
};

export const passwordDescription = (date: JSX.Element, time: JSX.Element) => {
    console.log(date, time);
    return {
        id: t("user.settings.security.passwordDescription"),
        defaultMessage: `Last updated ${{ date }} at ${{ time }}`,
    };
};

export const passw = (d: Date, militaryTime: boolean) => {
    return (
        <FormattedMessage
            id="user.settings.security.lastUpdated"
            defaultMessage="Last updated {date} at {time}"
            values={{
                date: (
                    <FormattedDate
                        value={d}
                        day="2-digit"
                        month="short"
                        year="numeric"
                    />
                ),
                time: (
                    <FormattedTime
                        value={d}
                        hour12={!militaryTime}
                        hour="2-digit"
                        minute="2-digit"
                    />
                ),
            }}
        />
    );
};

export const MfaTitle = {
    id: t("user.settings.security.MfaTitle"),
    defaultMessage: "Multi-factor authentication (MFA)",
};

export const MfaDescription = {
    id: t("user.settings.security.MfaDescription"),
    defaultMessage:
        "Adding multi-factor authentication will make your account more secure by requiring a code from your mobile device each time you sign in.",
};

export const AccessHistoryTitle = {
    id: t("user.settings.security.AccessHistoryTitle"),
    defaultMessage: "Access history",
};

export const AccessHistoryDesc = {
    id: t("user.settings.security.AccessHistoryDesc"),
    defaultMessage:
        "View a list of recent login and logout attempts, channel creations and deletions, account settings changes, or channel setting modifications made on your account.",
};

export const showAccessHistory = () => (
    <ToggleModalButton
        className="security_links"
        modalId="access_history"
        dialogType={AccessHistoryModal}
        id="viewAccessHistory"
    >
        <LocalizedIcon
            className="fa fa-clock-o"
            title={{
                id: t("user.settings.security.viewHistory.icon"),
                defaultMessage: "Access History Icon",
            }}
        />
        <FormattedMessage
            id="user.settings.security.viewHistory"
            defaultMessage="View Access History"
        />
    </ToggleModalButton>
);

export const ActiveSectionsTitle = {
    id: t("user.settings.security.ActiveSectionsTitle"),
    defaultMessage: "Active sessions",
};

export const ActiveSectionsDesc = {
    id: t("user.settings.security.ActiveSectionsDesc"),
    defaultMessage:
        "Sessions are created when you log in through a new browser on a device. You can view all active sessions and log out of any session you want to revoke.",
};
export const showActiveSections = () => (
    <ToggleModalButton
        className="security_links"
        modalId="activity_log"
        dialogType={ActivityLogModal}
        id="viewAndLogOutOfActiveSessions"
    >
        <LocalizedIcon
            className="fa fa-clock-o"
            title={{
                id: t("user.settings.security.logoutActiveSessions.icon"),
                defaultMessage: "Active Sessions Icon",
            }}
        />
        <FormattedMessage
            id="user.settings.security.ActiveSessions"
            defaultMessage="View Active Sessions"
        />
    </ToggleModalButton>
);
