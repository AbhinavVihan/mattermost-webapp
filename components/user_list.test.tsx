// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Provider} from 'react-redux';

import {TestHelper} from 'utils/test_helper';

import {renderWithIntl} from 'tests/react_testing_utils';
import store from 'stores/redux_store';

import UserList from './user_list';

describe('components/UserList', () => {
    test('should match default snapshot', () => {
        const props = {
            actionProps: {
                mfaEnabled: false,
                enableUserAccessTokens: false,
                experimentalEnableAuthenticationTransfer: false,
                doPasswordReset: jest.fn(),
                doEmailReset: jest.fn(),
                doManageTeams: jest.fn(),
                doManageRoles: jest.fn(),
                doManageTokens: jest.fn(),
                isDisabled: false,
            },
        };
        const wrapper = renderWithIntl(
            <Provider store={store}>
                <UserList {...props}/>,
            </Provider>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match default snapshot when there are users', () => {
        const User1 = TestHelper.getUserMock({id: 'id1'});
        const User2 = TestHelper.getUserMock({id: 'id2'});
        const props = {
            users: [
                User1,
                User2,
            ],
            actionUserProps: {},
            actionProps: {
                mfaEnabled: false,
                enableUserAccessTokens: false,
                experimentalEnableAuthenticationTransfer: false,
                doPasswordReset: jest.fn(),
                doEmailReset: jest.fn(),
                doManageTeams: jest.fn(),
                doManageRoles: jest.fn(),
                doManageTokens: jest.fn(),
                isDisabled: false,
            },
        };

        const wrapper = renderWithIntl(
            <Provider store={store}>
                <UserList {...props}/>,
            </Provider>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
