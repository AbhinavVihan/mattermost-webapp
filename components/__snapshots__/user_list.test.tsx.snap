// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/UserList should match default snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="more-modal__placeholder-row"
          data-testid="noUsersFound"
        >
          <p>
            No users found
          </p>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="more-modal__placeholder-row"
        data-testid="noUsersFound"
      >
        <p>
          No users found
        </p>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/UserList should match default snapshot when there are users 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="more-modal__row"
        >
          <span
            class="status-wrapper  "
          >
            <button
              class="RoundButton-dvlhqG gfRzmz style--none"
            >
              <span
                class="profile-icon "
              >
                <img
                  alt="some-user profile image"
                  class="Avatar Avatar-md"
                  loading="lazy"
                  src="/api/v4/users/id1/image"
                  tabindex="-1"
                />
              </span>
            </button>
          </span>
          <div
            class="more-modal__details"
            data-testid="userListItemDetails"
          >
            <div
              class="d-flex whitespace--nowrap"
            >
              <div
                class="more-modal__name"
              >
                <button
                  aria-label="someone"
                  class="user-popover style--none"
                >
                  Someone
                </button>
                 
                 
              </div>
            </div>
            <div
              class="more-modal__description"
            />
          </div>
          <div
            class="more-modal__actions"
            data-testid="userListItemActions"
          />
        </div>
        <div
          class="more-modal__row"
        >
          <span
            class="status-wrapper  "
          >
            <button
              class="RoundButton-dvlhqG gfRzmz style--none"
            >
              <span
                class="profile-icon "
              >
                <img
                  alt="some-user profile image"
                  class="Avatar Avatar-md"
                  loading="lazy"
                  src="/api/v4/users/id2/image"
                  tabindex="-1"
                />
              </span>
            </button>
          </span>
          <div
            class="more-modal__details"
            data-testid="userListItemDetails"
          >
            <div
              class="d-flex whitespace--nowrap"
            >
              <div
                class="more-modal__name"
                id="userListRowName1"
              >
                <button
                  aria-label="someone"
                  class="user-popover style--none"
                >
                  Someone
                </button>
                 
                 
              </div>
            </div>
            <div
              class="more-modal__description"
              id="userListRowEmail1"
            />
          </div>
          <div
            class="more-modal__actions"
            data-testid="userListItemActions"
          />
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="more-modal__row"
      >
        <span
          class="status-wrapper  "
        >
          <button
            class="RoundButton-dvlhqG gfRzmz style--none"
          >
            <span
              class="profile-icon "
            >
              <img
                alt="some-user profile image"
                class="Avatar Avatar-md"
                loading="lazy"
                src="/api/v4/users/id1/image"
                tabindex="-1"
              />
            </span>
          </button>
        </span>
        <div
          class="more-modal__details"
          data-testid="userListItemDetails"
        >
          <div
            class="d-flex whitespace--nowrap"
          >
            <div
              class="more-modal__name"
            >
              <button
                aria-label="someone"
                class="user-popover style--none"
              >
                Someone
              </button>
               
               
            </div>
          </div>
          <div
            class="more-modal__description"
          />
        </div>
        <div
          class="more-modal__actions"
          data-testid="userListItemActions"
        />
      </div>
      <div
        class="more-modal__row"
      >
        <span
          class="status-wrapper  "
        >
          <button
            class="RoundButton-dvlhqG gfRzmz style--none"
          >
            <span
              class="profile-icon "
            >
              <img
                alt="some-user profile image"
                class="Avatar Avatar-md"
                loading="lazy"
                src="/api/v4/users/id2/image"
                tabindex="-1"
              />
            </span>
          </button>
        </span>
        <div
          class="more-modal__details"
          data-testid="userListItemDetails"
        >
          <div
            class="d-flex whitespace--nowrap"
          >
            <div
              class="more-modal__name"
              id="userListRowName1"
            >
              <button
                aria-label="someone"
                class="user-popover style--none"
              >
                Someone
              </button>
               
               
            </div>
          </div>
          <div
            class="more-modal__description"
            id="userListRowEmail1"
          />
        </div>
        <div
          class="more-modal__actions"
          data-testid="userListItemActions"
        />
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
